staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -2580,12 +2582,12 @@ func describeSecret(secret *corev1.Secret) (string, error) {
 		w.Write(LEVEL_0, "\nType:\t%s\n", secret.Type)
 
 		w.Write(LEVEL_0, "\nData\n====\n")
-		for k, v := range secret.Data {
+		for _, k := range slices.Sorted(maps.Keys(secret.Data)) {
```

Comment:
Minor clarification: I understand in the issue that we wanted secrets to be displayed in "stable order", but I'm not quite sure what that means exactly. I'm guessing the sorting procedure is good enough here, but wondering if this needs to be more specific. I looked at the [docs for `slices.Sorted`](https://pkg.go.dev/slices#Sorted) doesn't say anything about stable ordering.

Reply:
I think I was a bit careless using the phrase "stable order" in the issue.

I have removed it now and replaced it with "predictable order". That is my intention with this fix.

Given the we are sorting a map, meaning keys cannot have duplicates, won't that make the concept of stable meaningless anyway?

Reply:
Yeah, I also thought about the fact that keys are unique in a map. With that said, another issue (at least for me) may be that it's not clear what order `slices.Sort` arranges (what I assume are) string keys. The tests in this PR don't seem to make assumptions around that exhaustively (and I'm not sure how much of that should be covered).

Reply:
The keys are strings and slices.Sort sorts them alphabetically.

Reply:
I raised a PR against your fork to parameterise the test and add more test cases.

Reply:
@rdgoite Thanks, but I don't see the point of testing a standard library function.

After my PR is merged, maybe you can add your refactoring as a separate PR against kubernetes, how does that sound? 🤗

Reply:
It's not really testing a standard library function. I think adding more test cases makes the intended behaviour of describe secret more explicit. Using the standard library function is an implementation detail that can change in the future for whatever reason. On the other end, the desired sorting behaviour can also change which means the tests need to make that explicit.

Essentially my PR is just your original test with additional test cases to make it clearer. Is your concern around having to go through a fresh round of code review? Apologies as this is my first attempt at contributing to this repo, but I'm not sure why simple additional test cases should require another round of PR process.

Reply:
@rdgoite thank you for pushing us forward for a better test coverage. I think, it is better to merge this PR as is and you can open a followup PR to increase the test coverage of this functionality. That would be simpler approach.

Reply:
@ardaguclu Ok. Is the plan to close kubectl#1735 and raise another ticket to add those test cases or are we merging this PR and leaving that ticket open for the new test cases?